tiledb/sm/array_schema/array_schema.cc

```diff
@@ -1696,36 +1694,19 @@ void ArraySchema::clear() {
   timestamp_range_ = std::make_pair(0, 0);
 }
 
-Status ArraySchema::generate_uri() {
-  std::string uuid;
-  RETURN_NOT_OK(uuid::generate_uuid(&uuid, false));
-
-  auto timestamp = utils::time::timestamp_now_ms();
-  timestamp_range_ = std::make_pair(timestamp, timestamp);
-  std::stringstream ss;
-  ss << "__" << timestamp_range_.first << "_" << timestamp_range_.second << "_"
-     << uuid;
-  name_ = ss.str();
-  uri_ =
-      array_uri_.join_path(constants::array_schema_dir_name).join_path(name_);
-
-  return Status::Ok();
-}
-
-Status ArraySchema::generate_uri(
-    const std::pair<uint64_t, uint64_t>& timestamp_range) {
-  std::string uuid;
-  RETURN_NOT_OK(uuid::generate_uuid(&uuid, false));
+void ArraySchema::generate_uri(
```

Comment:
Note that we could get rid of this function altogether. It is only called in a few places where a new constructor would remove the need for this. But not necessary to do in this PR.

Reply:
I noticed the same thing but the `clear()` functions in `ArraySchema` and `ArraySchemaEvolution` introduce a slight complication to this change, particularly with serialization. It seems it would be better to make both classes C.41 compliant, and remove `generate_uri` as a part of that work.